components/notification-sender-config/org.wso2.carbon.identity.notification.sender.tenant.config/src/main/java/org/wso2/carbon/identity/notification/sender/tenant/config/NotificationSenderManagementServiceImpl.java

```diff
@@ -551,4 +584,55 @@ private String getChannelTypeFromResource(Resource resource) {
                     .map(Attribute::getValue).orElse(DEFAULT_HANDLER_NAME);
         }
     }
+
+    /**
+     * Method to check whether the sender is allowed to delete.
+     *
+     * @param senderName    Name of the sender.
+     * @return  whether the sender is allowed to delete.
+     * @throws NotificationSenderManagementException If an error occurred while checking the sender.
+     */
+    private boolean checkAllowDelete(String senderName) throws NotificationSenderManagementException {
+
+        if (SENDERS.get(senderName) != null) {
+            String authenticatorId = new String(Base64.getUrlEncoder()
+                    .encode(SENDERS.get(senderName).getBytes(StandardCharsets.UTF_8)),
+                    StandardCharsets.UTF_8);
+            String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+            try {
+                ConnectedAppsResult appsResult =
+                        NotificationSenderTenantConfigDataHolder.getInstance()
+                        .getApplicationManagementService()
+                        .getConnectedAppsForLocalAuthenticator(authenticatorId, tenantDomain, 1 , 0);
+                // If there are any connected apps, the sender cannot be deleted.
+                if (appsResult.getApps().size() > 0) {
+                    return false;
+                }
+                Resource resource = NotificationSenderTenantConfigDataHolder.getInstance().getConfigurationManager()
+                        .getResource(MY_ACCOUNT_SMS_RESOURCE_TYPE, MY_ACCOUNT_SMS_RESOURCE_NAME);
+                String smsOtpEnabled = resource.getAttributes().stream().filter(attribute -> attribute
```

Comment:
Can the resource or the resource.getAttributes  be null ?

Reply:
If the resource is null, it will throw an exception that is handled. Added the validation for getting attributes.